### PR TITLE
Creating cheeps from main

### PR DESCRIPTION
**Pull request by JaSNOHo**
FEAT/Fix: Create Cheep works and 4 minor fixes:
- Log in page has no email box anymore.
- After registering you don’t need to confirm your email.
- Timestamps on cheeps show current time (Now instead of UtcNow).
- GitHub registering asks for username instead of email.

### CODE REVIEW

File: src/ChirpInfrastructure/CheepService.cs [hunk 1]
```diff
@@ -1,6 +1,9 @@
-using ChirpCore.DTOs;
+using System.Data;
```
Comment (HenrijetteOest) [inline src/ChirpInfrastructure/CheepService.cs:1]: usings System.Data er potentielt updateret og andet using fra main? eller så ikke, så er jeg bare lidt usikker på hvad system.data gør igen

File: src/ChirpInfrastructure/CheepRepository.cs [hunk 2]
```diff
@@ -120,6 +146,11 @@ public async Task<Author> GetAuthorFromUsernameAsync(string? Username)
 		{
 			throw new ArgumentNullException(Username);
 		}
-		return await _context.Authors.Include(a => a.Follows).Where(Author => Author.UserName == Username).FirstAsync();
+		return await _context.Authors
+		.Include(a => a.Follows)
+		.Where(author => author.UserName == Username)
+		.FirstOrDefaultAsync();
+
+		//return await _context.Authors.Include(a => a.Follows).Where(Author => Author.UserName == Username).FirstAsync();
```
Comment (HenrijetteOest) [inline src/ChirpInfrastructure/CheepRepository.cs:154]: slettes udkommenteret

File: src/ChirpInfrastructure/CheepService.cs [hunk 3]
```diff
@@ -71,4 +77,40 @@ public async Task<List<CheepDTO>> GetCheepsFromAuthorAsync(string author, int pa
 
         return _cheeps;
     }
+    public async Task<int> CreateCheepAsync(int userId, string userName, string text)
+    {
+        Console.WriteLine($"Creating cheep for user {userId} with text: {text}");
```
Comment (tomat-suppe) [inline src/ChirpInfrastructure/CheepService.cs:82]: Delete this later

File: src/ChirpWeb/Pages/CreatingCheep.cshtml.cs [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Error: {ex.Message}");
+                ModelState.AddModelError(string.Empty, "An error occurred while submitting your Cheep.");
+                return Page();
+            }
+        }
+
+        /*public async Task<IActionResult> OnPostAsync()
```
Comment (HenrijetteOest) [inline src/ChirpWeb/Pages/CreatingCheep.cshtml.cs:71]: slette udkommenteret?

File: src/ChirpInfrastructure/CheepService.cs [hunk 5]
```diff
@@ -71,4 +77,40 @@ public async Task<List<CheepDTO>> GetCheepsFromAuthorAsync(string author, int pa
...
+
+        if (author != null)
+        {
+            author.Cheeps.Add(newCheep);
+        }
+
+        // Add the Cheep to the database context
+        await _context.Cheeps.AddAsync(newCheep);
+        await _context.SaveChangesAsync();
+        Console.WriteLine("Cheep successfully saved to database.");
```
Comment (tomat-suppe) [inline src/ChirpInfrastructure/CheepService.cs:113]: delete later

File: src/ChirpWeb/Areas/Identity/Pages/Account/ExternalLogin.cshtml.cs [hunk 6]
```diff
@@ -83,8 +83,8 @@ public class InputModel
             ///     directly from your code. This API may change or be removed in future releases.
             /// </summary>
             [Required]
-            [EmailAddress]
-            public string Email { get; set; }
+            //[EmailAddress]
```
Comment (tomat-suppe) [inline src/ChirpWeb/Areas/Identity/Pages/Account/ExternalLogin.cshtml.cs:86]: remove later?

File: src/ChirpWeb/Areas/Identity/Pages/Account/Login.cshtml [hunk 7]
```diff
@@ -19,13 +19,14 @@
                     <label asp-for="Input.UserName" class="form-label">UserName</label>
                     <span asp-validation-for="Input.UserName" class="text-danger"></span>
                 </div>
-                <div class="form-floating mb-3">
+                @*<div class="form-floating mb-3">
```
Comment (tomat-suppe) [inline src/ChirpWeb/Areas/Identity/Pages/Account/Login.cshtml:22]: remove later?

File: src/ChirpWeb/Areas/Identity/Pages/Account/Login.cshtml.cs [hunk 8]
```diff
@@ -73,9 +73,9 @@ public class InputModel
             ///     This API supports the ASP.NET Core Identity default UI infrastructure and is not intended to be used
             ///     directly from your code. This API may change or be removed in future releases.
             /// </summary>
-            [Required]
-            [EmailAddress]
-            public string Email { get; set; }
+            //[Required]
```
Comment (tomat-suppe) [inline src/ChirpWeb/Areas/Identity/Pages/Account/Login.cshtml.cs:76]: delete later

File: src/ChirpWeb/Pages/CreatingCheep.cshtml [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+@model ChirpWeb.Pages.CreateCheepModel
+@{
+    ViewData["Title"] = "Create Cheep";
+    Layout = "_Layout";
+}
+
+<h2>Create a New Cheep</h2>
+
+<form method="post">
+    @Html.AntiForgeryToken()
```
Comment (tomat-suppe) [inline src/ChirpWeb/Pages/CreatingCheep.cshtml:11]: This is a security feature to help protect your application against cross-site request forgery.

Source: https://stackoverflow.com/questions/44783702/what-is-the-use-of-html-antiforgerytoken

File: src/ChirpWeb/Pages/CreatingCheep.cshtml.cs [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+        private readonly UserManager<Author> _userManager;
+        public CreateCheepModel(ICheepRepository repository, UserManager<Author> userManager)
+        {
+            _repository = repository;
+            _userManager = userManager;
+            CheepText = "";
+        }
+        [BindProperty]
+        [Required(ErrorMessage = "Please enter a message for your Cheep.")]
+        [StringLength(280, ErrorMessage = "Cheep cannot exceed 280 characters.")]
```
Comment (tomat-suppe) [inline src/ChirpWeb/Pages/CreatingCheep.cshtml.cs:23]: Hov, det her skulle være 160

File: src/ChirpWeb/Pages/CreatingCheep.cshtml.cs [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+        public async Task<IActionResult> OnPostAsync()
+        {
+            if (!ModelState.IsValid)
+            {
+                return Page();
+            }
+
+            try
+            {
+                /*int userId = 0;
```
Comment (tomat-suppe) [inline src/ChirpWeb/Pages/CreatingCheep.cshtml.cs:36]: delete later

File: src/ChirpWeb/Pages/CreatingCheep.cshtml.cs [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+
+                if (User.Identity.IsAuthenticated)
+                {
+                    userId = int.Parse(User.FindFirst("sub")?.Value ?? "0");
+                    userName = User.Identity.Name ?? "Anonymous";
+                }*/
+
+                //Console.WriteLine($"Attempting to create Cheep by userId: {userId}, userName: {userName}");
+                Author author = await _userManager.FindByNameAsync(User.Identity.Name); // await _userManager.FindByIdAsync(userId.ToString()) ??
+                // Create the Cheep object
```
Comment (tomat-suppe) [inline src/ChirpWeb/Pages/CreatingCheep.cshtml.cs:47]: har vi det her i cheepservice?

**Review by HenrijetteOest — APPROVED** [review]
lgtm 🥉 :3

File: src/ChirpWeb/Pages/CreatingCheep.cshtml.cs [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+                // Save the Cheep using the repository
+                await _repository.AddCheepAsync(newCheep);
+
+                //Console.WriteLine("Cheep created successfully!");
+                return LocalRedirect("/");
+                //return RedirectToPage(); // Redirect after success
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Error: {ex.Message}");
```
Comment (tomat-suppe) [inline src/ChirpWeb/Pages/CreatingCheep.cshtml.cs:65]: delete later

**Review by tomat-suppe — APPROVED** [review]
looks good guys! we need to remember to remove out-commented code later (I've added some comments) and change cheep length restriction to 160 :)

File: src/ChirpInfrastructure/CheepService.cs (hunk 1 above)
Comment (HenrijetteOest) [reply src/ChirpInfrastructure/CheepService.cs:1]: nevermind, Hvad er det System.Data gør igen? :333333

File: src/ChirpWeb/Pages/CreatingCheep.cshtml.cs (hunk 13 above)
Comment (Thyme-Cat) [reply src/ChirpWeb/Pages/CreatingCheep.cshtml.cs:56]: Gonna keep this for error handling

File: src/ChirpInfrastructure/CheepService.cs (hunk 3 above)
Comment (Thyme-Cat) [reply src/ChirpInfrastructure/CheepService.cs:84]: Keeping this for now, to make error handling easier if we need it later (this method is also not called at the moment)

File: src/ChirpWeb/Pages/CreatingCheep.cshtml.cs (hunk 12 above)
Comment (Thyme-Cat) [reply src/ChirpWeb/Pages/CreatingCheep.cshtml.cs]: Ja, men metoden kaldes ikke og bør rykkes til CheepRepository. Så indtil videre bliver den her.

**Review by llsa-itu — APPROVED** [review]

File: src/ChirpInfrastructure/CheepService.cs (hunk 1 above)
Comment (tomat-suppe) [reply src/ChirpInfrastructure/CheepService.cs:1]: https://learn.microsoft.com/en-us/dotnet/api/system.argumentnullexception?view=net-9.0

System.Data er det library vi bruger til argumentnullexception ;3

File: src/ChirpWeb/Areas/Identity/Pages/Account/Login.cshtml (hunk 7 above)
Comment (Thyme-Cat) [reply src/ChirpWeb/Areas/Identity/Pages/Account/Login.cshtml]: removed

File: src/ChirpWeb/Areas/Identity/Pages/Account/ExternalLogin.cshtml.cs (hunk 6 above)
Comment (Thyme-Cat) [reply src/ChirpWeb/Areas/Identity/Pages/Account/ExternalLogin.cshtml.cs]: removed

File: src/ChirpWeb/Areas/Identity/Pages/Account/Login.cshtml.cs (hunk 8 above)
Comment (Thyme-Cat) [reply src/ChirpWeb/Areas/Identity/Pages/Account/Login.cshtml.cs]: removed

File: src/ChirpWeb/Pages/CreatingCheep.cshtml.cs (hunk 10 above)
Comment (Thyme-Cat) [reply src/ChirpWeb/Pages/CreatingCheep.cshtml.cs]: Fikset

File: src/ChirpWeb/Pages/CreatingCheep.cshtml.cs (hunk 4 above)
Comment (Thyme-Cat) [reply src/ChirpWeb/Pages/CreatingCheep.cshtml.cs]: removed

**Pull request merged**